### DISCUSSION TITLE
obs connect only works locally, not on a different server

### DISCUSSION THREAD

**Discussion by bguan2020** [obs-websocket-community-projects/obs-websocket-js#321] [Q&A]
i've been successfully connecting to my obs websocket via running this locally on the same ip address:
```
const obs = new OBSWebSocket();
  obs
    .connect(`ws://${obs_IP_address}`, obs_password)
```

However, the second I try running this code on a remote server, it returns this error:
```
connect ETIMEDOUT my_obs_IP_address
```

Can this only work locally or am I doing something wrong? What are my alternatives for updating my OBS via a remote server website api request?

**Comment by ConstantinAmurov** [comment]
As I understand, the connection cannot be established. Try first of all and ping  `my_obs_IP_address` , and try make a successful connection first of all.
